bool_slice.go

```diff
@@ -0,0 +1,145 @@
+package pflag
+
+import (
+	"encoding/csv"
+	"io"
+	"strconv"
+	"strings"
+)
+
+// -- boolSlice Value
+type boolSliceValue struct {
+	value   *[]bool
+	changed bool
+}
+
+func newBoolSliceValue(val []bool, p *[]bool) *boolSliceValue {
+	bsv := new(boolSliceValue)
+	bsv.value = p
+	*bsv.value = val
+	return bsv
+}
+
+// Set converts, and assigns, the comma-separated boolean argument string representation as the []bool value of this flag.
+// If Set is called on a flag that already has a []bool assigned, the newly converted values will be appended.
+func (s *boolSliceValue) Set(val string) error {
+
+	// remove all quote characters
+	rmQuote := strings.NewReplacer(`"`, "", `'`, "", "`", "")
+	r := csv.NewReader(strings.NewReader(rmQuote.Replace(val)))
+
+	// read flag arguments with CSV parser
+	boolStrSlice, err := r.Read()
+	if err != nil && err != io.EOF {
```

Comment:
I must have overlooked those CSV utility functions. You did mention them before. Sorry about that. I use those functions in my latest new commit.

If you think this design is acceptable, I can go back and handle the uintSliceValue type the same way if you like. I can also do the other *SliceValue types with the same design, but I would prefer not to do it on this pull request.